src/ui/component/dateInput/dateInput.tsx

```diff
@@ -0,0 +1,55 @@
+import { useMemo, type FC } from 'react'
+import { useTranslation } from 'react-i18next'
+import { Icon } from '@/ui/component/icon/icon'
+import './dateInput.scss'
+import type { LimitedDateTimeFormatOptions } from '@/util/i18n/i18n'
+import { localizedDateFormatter, formatDateToPattern } from '@/util/i18n/i18n'
+
+type DateInputProps = {
+  date?: Date
+}
+
+export const DateInput: FC<DateInputProps> = ({ date }) => {
+  const { i18n } = useTranslation()
+  const locale = i18n.language
+
+  const patternMap = useMemo(
```

Comment:
As I said below, why introducing `Map` notion here?

Reply:
My bad, it was a naming mistake, some details were provided in [this answer](https://github.com/okp4/dataverse-portal/pull/460#discussion_r1321660526) from your previous feedback.